bindgen/src/utils.ts

```diff
@@ -76,3 +76,9 @@ export function posixRelativePath(from: string, to: string): string {
 export function toString(node: Node): string {
   return ASTBuilder.build(node);
 }
+
+const captialPattern = /([A-Z])/g;
+
+export function makeSnakeCase(s: string): string {
+  return s.replace(captialPattern, "_$1").toLowerCase();
```

Comment:
This has a few weird cases, which may be acceptable:

`convertAPIKey` -> `convert_a_p_i_key`
`SomeFunction` -> `_some_function`

Reply:
Good point.  I think we to specify that it's not at the beginning. And add `+` to the capital match.  That should solve the issue.  I'll add a test and add your examples as test cases.